File: app/assets/javascripts/SubDetail1.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const SubDetail1Schema = DataElementSchema({
  productOrService: Any,
  modifier: Any,
  quantity: Any,
  unitPrice: Any,
  factor: Any,
  net: Any,
  noteNumber: Any,
  adjudication: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'SubDetail1' },

});

module.exports.SubDetail1Schema = SubDetail1Schema;
module.exports.SubDetail1 = mongoose.model('SubDetail1', SubDetail1Schema);

File: app/models/qdm/sub_detail1.rb
module QDM
  # app/models/qdm/sub_detail1.rb
  class SubDetail1 < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :productOrService
    field :modifier
    field :quantity
    field :unitPrice
    field :factor
    field :net
    field :noteNumber
    field :adjudication
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
